src/views/Rules/Rule/index.jsx

```diff
@@ -117,10 +132,109 @@ export default function Rule({ isNewRule, ...props }) {
     setDateTimePickerError(error);
   };
 
-  // TODO: Add logic for actions
-  const handleScheduleChangeDelete = () => {};
-  const handleCreateRule = () => {};
-  const handleUpdateRule = () => {};
+  const handleScheduleChangeDelete = async () => {
+    const { error } = await deleteSC({
+      scId: rule.sc_id,
+      scDataVersion: rule.sc_data_version,
+    });
+
+    if (!error) {
+      props.history.push('/rules');
+    }
+  };
+
+  const handleCreateRule = async () => {
+    const now = new Date();
+    const when =
+      scheduleDate >= now ? scheduleDate.getTime() : now.getTime() + 5000;
+    const data = {
+      alias: rule.alias,
+      backgroundRate: rule.backgroundRate,
+      buildID: rule.buildID,
+      buildTarget: rule.buildTarget,
+      channel: rule.channel,
+      comment: rule.comment,
+      distVersion: rule.distVersion,
+      distribution: rule.distribution,
+      fallbackMapping: rule.fallbackMapping,
+      headerArchitecture: rule.headerArchitecture,
+      instructionSet: rule.instructionSet,
+      jaws: rule.jaws,
+      locale: rule.locale,
+      mapping: rule.mapping,
+      memory: rule.memory,
+      mig64: rule.mi64,
+      osVersion: rule.osVersion,
+      priority: rule.priority,
+      product: rule.product,
+      update_type: rule.update_type,
+      version: rule.version,
+    };
+    const { error } = await addSC({
+      change_type: 'insert',
+      when,
+      ...data,
+    });
+
+    if (!error) {
+      props.history.push('/rules');
+    }
+  };
+
+  const handleUpdateRule = async () => {
+    const now = new Date();
+    const when =
+      scheduleDate >= now ? scheduleDate.getTime() : now.getTime() + 5000;
+    const data = {
+      alias: rule.alias,
+      backgroundRate: rule.backgroundRate,
+      buildID: rule.buildID,
```

Comment:
Should this be `build_ID`?

Reply:
I don't think so. I just took a look and the only place (in both the frontend and backend) that uses the underscore is https://github.com/mozilla-frontend-infra/balrog-ui/blob/master/src/utils/constants.js#L16. I'll update that, instead.